### PR TITLE
Update golang library dependencies to address possible vulnerabilities

### PR DESCRIPTION
**Pull request by ShellyKa13**
**What this PR does / why we need it**:
This PR Bumps several go packages to address possible vulnerabilitie

* Bump github.com/opencontainers/runc v1.1.5 -> v1.1.12
    This should fix the following CVE:
    CVE-2024-21626

* Bump github.com/docker/docker v23.0.2+incompatible -> v25.0.5+incompatible 
    This should fix several CVEs:
    GHSA-jq35-85cj-fj4p, GHSA-mq39-4gv4-mvpx, CVE-2024-24557, CVE-2023-28842
    CVE-2023-28841, CVE-2023-28840
    
* Remove replace of github.com/aws/aws-sdk-go => github.com/aws/aws-sdk-go v1.15.77    
    We want to have the update version of aws-sdk-go to fix this issues:
    cve-2020-8911, cve-2022-2582, GHSA-76wf-9vgp-pj7w, cve-2020-8912

-->
```release-note
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -81,7 +81,7 @@ require (
 	github.com/containers/storage v1.46.0 // indirect
 	github.com/davecgh/go-spew v1.1.2-0.20180830191138-d8f796af33cc // indirect
 	github.com/docker/distribution v2.8.2+incompatible // indirect
-	github.com/docker/docker v23.0.2+incompatible // indirect
+	github.com/docker/docker v25.0.5+incompatible // indirect
```
Comment (akalenyu) [inline go.mod:84]: It's possible that bumping our container images library `github.com/containers/image/v5` to something newer would automatically bump the runc/docker ones as well.
I see cri-o main uses `v5.30.0` - https://github.com/cri-o/cri-o/pull/8133

**Review by akalenyu — COMMENTED** [review]
Looks good maybe just bump `github.com/containers/image/v5` to something newer?

File: go.mod (hunk 1 above)
Comment (ShellyKa13) [reply go.mod:84]: I tried that and it didnt bump it to new enough version without the CVE..
Also I didnt see we have cri-o in our go.mod..

Comment (ShellyKa13) [reply go.mod:84]: hmm now I see it should be the same version, let me try again.

Comment (akalenyu) [reply go.mod:84]: Right I see, makes sense

**Pull request merged**